### PR TITLE
[Tutorials] Update the SDG tutorial and expose the inference endpoint

### PR DESCRIPTION
**Pull request by Maghoumi**
## Description
This PR ensures that users can run the PEFT SDG tutorial using arbitrary API endpoints by exposing the URL that is used for synthetic data generation.

## Checklist
- [X] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [X] New or Existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: tutorials/peft-curation-with-sdg/README.md [hunk 1]
```diff
@@ -48,45 +48,49 @@ showcased in this code:
...
 
-# Run with synthetic data generation and semantic dedeuplication
+# Running with synthetic data generation and semantic dedeuplication using
+# an external LLM inference endpoint located at "https://api.example.com/v1/chat/completions"
+# and the model called "my-llm-model" that is served at that endpoint:
 python tutorials/peft-curation-with-sdg/main.py \
-    --api-key YOUR_BUILD.NVIDIA.COM_API_KEY \
+    --synth-gen-endpoint https://api.example.com/v1/chat/completions \
+    --synth-gen-model my-llm-model \
+    --api-key API_KEY_FOR_EXAMPLE.COM \
```
Comment (ryantwolf) [inline tutorials/peft-curation-with-sdg/README.md:70]: Nit: Might just be me, but `API_KEY_FOR_EXAMPLE.COM` makes me think that the API key should end in `.com`, which it shouldn't. Maybe make it `API_KEY_FOR_ENDPOINT` instead?

**Review by ryantwolf — CHANGES_REQUESTED** [review]
I just have the smallest of nits and we should be good. Also, the DCO check seems to be failing so follow the point 3 in our [contributing guide pr guidelines](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md#pull-requests-pr-guidelines) to fix it.

File: tutorials/peft-curation-with-sdg/README.md (hunk 1 above)
Comment (Maghoumi) [reply tutorials/peft-curation-with-sdg/README.md]: Done.

**Comment by Maghoumi** [comment]
Also fixed the DCO check. Looks like I forgot the `-sS` flag when committing.

**Pull request merged**